packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts

```diff
@@ -280,14 +282,20 @@ export const a11y: AuditRuleWithSelector[] = [
 		selector: 'a[href]:is([href=""], [href="#"], [href^="javascript:" i])',
 	},
 	{
-		code: 'a11y-label-has-associated-control',
-		title: '`label` tag should have an associated control and a text content.',
+		code: 'a11y-invalid-label',
+		title: '`label` element should have an associated control and a text content.',
 		message:
-			'The `label` tag must be associated with a control using either `for` or having a nested input. Additionally, the `label` tag must have text content.',
-		selector: 'label:not([for])',
+			'The `label` element must be associated with a control using either `for` or having a nested form element. Additionally, the `label` element must have text content.',
+		selector: 'label',
 		match(element) {
-			const inputChild = element.querySelector('input');
-			if (!inputChild?.textContent) return true;
+			// Label must be associated with a control
+			const hasFor = element.hasAttribute('for');
```

Comment:
Should we also be checking that the `for` attribute element actually exists? Or does another audit take care of that?

Reply:
Not sure I understand what you mean. The line you highlighted checks for `for`?

Reply:
Oh, I just got it, that the element the for is for exists, that's neat. We could do another rule for that, this rule is already a bit overloaded!